From 07eb2f07a6ec24deb68980f1544fff7a206479fc Mon Sep 17 00:00:00 2001
From: Lukas Prediger <lukas.prediger@csc.fi>
Date: Thu, 1 Aug 2024 09:59:36 +0000
Subject: [PATCH 2/3] Check ROCM install for libhipblas.so* instead of
 libhipblas.so.2*

AMDGetGPUInfo previously required libhipblas.so.2* to be present
in the ROCM installation or it would ignore availalbe GPUs.
libhipblas 2 is only available starting from ROCm v6.
Ollama appears to be compiling and running fine with older versions
of ROCm (tested for 5.6.1) which come with libhipblas.so.1*.

This commit relaxes the filter in ROCmLibGlobs to accept an
libhipblas.so*.
---
 gpu/amd_linux.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gpu/amd_linux.go b/gpu/amd_linux.go
index 66862ea5..e4a010f7 100644
--- a/gpu/amd_linux.go
+++ b/gpu/amd_linux.go
@@ -41,7 +41,7 @@ const (
 
 var (
 	// Used to validate if the given ROCm lib is usable
-	ROCmLibGlobs          = []string{"libhipblas.so.2*", "rocblas"} // TODO - probably include more coverage of files here...
+	ROCmLibGlobs          = []string{"libhipblas.so*", "rocblas"} // TODO - probably include more coverage of files here...
 	RocmStandardLocations = []string{"/opt/rocm/lib", "/usr/lib64"}
 )
 
-- 
2.34.1

